src/python/pants_release/start_release.py

```diff
@@ -200,21 +257,84 @@ def splice_into_file(release_info: ReleaseInfo, formatted: Formatted) -> None:
         # default content if the file doesn't exist yet
         existing_contents = f"# {release_info.slug} Release Series\n"
 
-    file_name.write_text(splice(existing_contents, formatted.external))
+    file_name.write_text(splice(str(release_info.version), existing_contents, formatted.external))
 
 
-def main() -> None:
-    args = create_parser().parse_args()
-    release_info = ReleaseInfo.determine(args.new)
+def update_changelog(release_info: ReleaseInfo) -> Formatted:
     branch_sha = git_fetch(release_info.branch)
     date = datetime.date.today()
-    entries = [prepare_sha(sha) for sha in relevant_shas(args.prior, branch_sha)]
+    entries = [prepare_sha(sha) for sha in relevant_shas(branch_sha)]
 
     formatted = format_notes(release_info, entries, date)
     splice_into_file(release_info, formatted)
-    print(f"\nCommit {release_info.notes_file_name()} and create a PR.\n\n{formatted.internal}")
+
+    return formatted
+
+
+def update_contributors(release_info: ReleaseInfo) -> None:
+    if release_info.branch == "main":
+        CONTRIBUTORS_PATH.write_text(
+            "Created as part of the release process.\n\n+ "
+            + "\n+ ".join(sorted_contributors(git_range="HEAD"))
+            + "\n"
+        )
+
+
+def update_version(release_info: ReleaseInfo) -> None:
+    if release_info.branch == "main":
+        VERSION_PATH.write_text(f"{release_info.version}\n")
+
+
+def commit_and_pr(
+    repo: github.Repository.Repository,
+    release_info: ReleaseInfo,
+    formatted: Formatted,
+    release_manager: str,
+) -> None:
+    title = f"Prepare {release_info.version}"
+    branch = f"automation/release/{release_info.version}"
+
+    git("checkout", "-b", branch)
+    git("add", str(VERSION_PATH), str(CONTRIBUTORS_PATH), str(release_info.notes_file_name()))
+    git("commit", "-m", title)
+    git("push", "origin", "HEAD")
+
+    pr = repo.create_pull(
+        title=title,
+        body=formatted.internal or "(No internal changes.)",
+        base="main",
+        head=branch,
+    )
+    pr.add_to_assignees(release_manager)
+    pr.add_to_labels("automation:release-prep", "category:internal")
+    pr.create_issue_comment(PR_COMMENT_BODY)
+
+
+def main() -> None:
+    args = create_parser().parse_args()
+    logging.basicConfig(level=args.log_level)
+
+    if args.publish:
+        # fail faster
+        gh_repo = github_repo()
+    else:
+        gh_repo = None
```

Comment:
Ternarys are easier to grok IMO
```suggestion
    # Declared up here, so we can fail faster
    gh_repo = github_repo() if args.publish else None
```

Reply:
Eh, I think this is at best a tiny improvement, and, for me, feels worse when there's side-effects/expensive processing in one branch.

That said, I'll make the change since you felt strongly enough to comment.

Reply:
I'm like 95% sure the Python ternary doesn't evaluate the "wrong" side of the expression.

Reply:
Yeah, they don't, it's more conceptually I find it clearer to have an actual code block for heavy-weight code, since it, e.g., gives a better spot to put comments on each branch.